root/static/scripts/edit/externalLinks.js

```diff
@@ -28,40 +30,41 @@ import * as URLCleanup from './URLCleanup';
 import validation from './validation';
 
 type LinkStateT = {
-  url: string,
-  type: number | null,
   relationship: number | string | null,
+  type: number | null,
+  url: string,
   video: boolean,
 };
 
 type LinksEditorProps = {
+  errorObservable: (boolean) => void,
+  initialLinks: Array<LinkStateT>,
   sourceType: string,
   typeOptions: Array<React.Element<'option'>>,
-  initialLinks: Array<LinkStateT>,
-  errorObservable: (bool) => void,
 };
 
 type LinksEditorState = {
   links: Array<LinkStateT>,
 };
 
-export class ExternalLinksEditor extends React.Component<LinksEditorProps, LinksEditorState> {
+export class ExternalLinksEditor
+  extends React.Component<LinksEditorProps, LinksEditorState> {
   constructor(props: LinksEditorProps) {
     super(props);
     this.state = {links: withOneEmptyLink(props.initialLinks)};
   }
 
   setLinkState(index: number,
-               state: $Shape<LinkStateT>,
-               callback?: () => void) {
+    state: $Shape<LinkStateT>,
+    callback?: () => void) {
```

Comment:
Heh, this looks weird now too, it was more readable before. Putting them all on their own line as suggested for `getFormData` would make sense to me.